Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -671,6 +664,20 @@ public async Task<SandboxedProcessPipExecutionResult> RunAsync(CancellationToken
             }
         }
 
+        private bool ShouldSandboxedProcessExecuteExternal
+            => // Execution mode is external
+               m_sandboxConfig.AdminRequiredProcessExecutionMode.ExecuteExternal()

```

Comment:
>m_sandboxConfig.AdminRequiredProcessExecutionMode.ExecuteExternal() [](start = 15, length = 67)

Should it error in the case where process admin requires external execution but it cannot be launched externally for other reasons?

Reply:
The pip is annotated with `privilegeLevel: Admin` field to denote that the pip needs to run as admin. The sandboxed configuration controls where to execute the pip. If, for example, it's set to external/VM, but then BuildXL is run on Unix or container is enabled, then it would be annoying if we failed the pip. I can log a warning instead of error in this case.

---
In reply to: [288712689](https://github.com/microsoft/BuildXL/pull/382#discussion_r288712689) [](ancestors = 288712689)